### PR TITLE
[GetMetaTag] Added the Mix_GetMetaTag function.

### PR DESCRIPTION
**Pull request by SNSTRUTHERS**
A quick-and-dirty way to get meta tags from a Mix_Music pointer. I added it for a project I'm working on.

### CODE REVIEW

**Comment by Wohlstand** [comment]
@sezero , do you think it's good yourself?

**Comment by Wohlstand** [comment]
A think looks interesting :fox_face: 
However, please remove "snstruthers" note and leave comments just "Mixer-X" :fox_face:

**Comment by SNSTRUTHERS** [comment]
Yeah yeah. I wasn't thinking of doing a pull request at first, hence the named comments.

I'm away from my computer at the moment (am typing on phone), so I'll have to do that later.

File: src/music.h [hunk 1]
```diff
@@ -23,6 +23,9 @@
 #ifndef MUSIC_H_
 #define MUSIC_H_
 
+#define LOOP_DEBUG_PRINTOUT 1
+/*#define LOOP_DEBUG_PRINTOUT 0*/
```
Comment (Wohlstand) [inline src/music.h:27]: You don't need to keep this here. Please make the CMake side option to turn this on/off :eyes:

Comment (SNSTRUTHERS) [reply src/music.h:27]: 👍

**Comment by Wohlstand** [comment]
> Yeah yeah. I wasn't thinking of doing a pull request at first, hence the named comments.
> 
> I'm away from my computer at the moment (am typing on phone), so I'll have to do that later.

No worry, make any necessary changes once you can :wink:

**Comment by SNSTRUTHERS** [comment]
> No worry, make any necessary changes once you can :wink: 

Thanks :^)

File: src/codecs/music_gme.c [hunk 2]
```diff
@@ -547,6 +547,7 @@ Mix_MusicInterface Mix_MusicInterface_GME =
     NULL,   /* LoopEnd [MIXER-X]*/
     NULL,   /* LoopLength [MIXER-X]*/
     GME_GetMetaTag,/* GetMetaTag [MIXER-X]*/
+    NULL,   /* GetUserTa [MIXER-X-snstruthers]*/
```
Comment (Wohlstand) [inline src/codecs/music_gme.c:550]: A typo: `GetUserTa`. Did you meant the `GetUserTag`?

Comment (SNSTRUTHERS) [reply src/codecs/music_gme.c:550]: LOL

**Comment by sezero** [comment]
> @sezero , do you think it's good yourself?

The debug print stuff pollutes the whole thing badly.

Apart from that, I think it needs explanation as to how it differs from other metadata functionality, whether it can be merged with the existing metatags stuff etc.

**Comment by SNSTRUTHERS** [comment]
> The debug print stuff pollutes the whole thing badly.
> 
> 
> 
> Apart from that, I think it needs explanation as to how it differs from other metadata functionality, whether it can be merged with the existing metatags stuff etc.

That was me forgetting to turn that flag off. I'll change it to a CMAKE flag later.

**Comment by Wohlstand** [comment]
> > @sezero , do you think it's good yourself?
> 
> The debug print stuff pollutes the whole thing badly.
> 
> Apart from that, I think it needs explanation as to how it differs from other metadata functionality, whether it can be merged with the existing metatags stuff etc.

That would require some sort of associative array at the to keep all meta-tags at the `Mix_MusicMetaTags` for a quick access without of necessary to make one more extra call entry inside of `Mix_MusicInterface`. That should make change being majorely smaller and be more universal. I.e. While loading song, put all tags into `Mix_MusicMetaTags` structure's field (all unhandled tags should use a generic assicoative key-value array), and when requesting an every tag, just access the structure instance instead of calling API of a codec.

**Comment by SNSTRUTHERS** [comment]
> That would require some sort of associative array at the to keep all meta-tags at the `Mix_MusicMetaTags` for a quick access without of necessary to make one more extra call entry inside of `Mix_MusicInterface`. That should make change being majorely smaller and be more universal. I.e. While loading song, put all tags into `Mix_MusicMetaTags` structure's field (all unhandled tags should use a generic assicoative key-value array), and when requesting an every tag, just access the structure instance instead of calling API of a codec.

I was thinking of doing that, but
1) I don't know of the intricacies regarding most of these file formats and how they store metadata.
2) I didn't want to add a hashmap library to the codebase.
3) The thing I'm working on that makes use of this extra feature only really needs to read the tag once, and that's it. Wasn't really worried about efficiency; just getting something that worked done quick.

**Comment by Wohlstand** [comment]
1. Don't worry on that, I think, I'll make the support for other formats myself. Except of MP3 tags are complex and messy (for them I developed the most basic system only which is needed for important well-known tags).
2. You don't need that, just make a simple fixed-size array (for example, with 200 entries), and store inside the `char *key` and `char*value` fields. If structure is null, it's an end of array. When searching an entry, just fetch from begin up to first null key and compare it. When it matches (case-insensitive), return a value, otherwise, return null when nothing was found.
3. In 2 I had to explain that it's fine. The main point of me and @sezero is that it's possible to make the code itself being more compact.

**Comment by SNSTRUTHERS** [comment]
Well at least I gave you guys an idea or starting off point lol

**Comment by Wohlstand** [comment]
I do think, the idea is good, but it should be implemented differently:
- make the cache of all meta-tags on the loading stage
- access every meta-tag from the cache directly, don't request tags from the library every time, that is a waste of computing time

I'll close this as it became stale, but I will make the task issue to implement the ability once upon.

**Pull request closed without merging**